tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py

```diff
@@ -1195,20 +1234,33 @@ def test__get_fn_arguments_with_defaults(self):
                 (sklearn.tree.DecisionTreeClassifier.__init__, 13),
                 (sklearn.pipeline.Pipeline.__init__, 1),
             ]
-        else:
+        elif sklearn_version < "0.22":
             fns = [
                 (sklearn.ensemble.RandomForestRegressor.__init__, 16),
                 (sklearn.tree.DecisionTreeClassifier.__init__, 13),
                 (sklearn.pipeline.Pipeline.__init__, 2),
             ]
+        elif sklearn_version < "0.23":
+            fns = [
+                (sklearn.ensemble.RandomForestRegressor.__init__, 18),
+                (sklearn.tree.DecisionTreeClassifier.__init__, 14),
+                (sklearn.pipeline.Pipeline.__init__, 2),
+            ]
+        else:
+            fns = [
+                (sklearn.ensemble.RandomForestRegressor.__init__, 0),
+                (sklearn.tree.DecisionTreeClassifier.__init__, 0),
+                (sklearn.pipeline.Pipeline.__init__, 0),
+            ]
 
         for fn, num_params_with_defaults in fns:
             defaults, defaultless = self.extension._get_fn_arguments_with_defaults(fn)
             self.assertIsInstance(defaults, dict)
             self.assertIsInstance(defaultless, set)
             # check whether we have both defaults and defaultless params
             self.assertEqual(len(defaults), num_params_with_defaults)
-            self.assertGreater(len(defaultless), 0)
+            if sklearn_version < "0.23":
+                self.assertGreater(len(defaultless), 0)
```

Comment:
You could add an else here to check that the number of defaultless hyperparameters is actually zero.

Reply:
Done.

Reply:
Sorry, I don't see the 'else' here. What I meant was:
```python
            if sklearn_version < "0.23":
                self.assertGreater(len(defaultless), 0)
            else:
                self.assertEqual(len(defaultless), 0)

Reply:
Aah, I should have mentioned.

So, since we moved to `inspect.signature` across sklearn versions as done [here](https://github.com/openml/openml-python/blob/023ac406af607da934ce15c0933d14b8665a19c5/openml/extensions/sklearn/extension.py#L997), we don't need [this](https://github.com/openml/openml-python/blob/023ac406af607da934ce15c0933d14b8665a19c5/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L1262) if-else anymore.

Reply:
I see.